### PR TITLE
Setup of rails app

### PR DESCRIPTION
**Pull request by monogut**
Added angular setup, removed turbo links, added gems for translation, made model relations between vocab and articles (belongs_to, has_many)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
+gem 'rails', '4.2.3'
+# Mongo DB
+#gem 'mongoloid'
+
+# Postgresql
+gem 'pg'
+# Use sqlite3 as the database for Active Record
+gem 'sqlite3'
```
Comment (f3r) [inline Gemfile]: you don't need sqlite3

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+# Use Unicorn as the app server
+# gem 'unicorn'
+
+# Use Capistrano for deployment
+# gem 'capistrano-rails', group: :development
+
+group :development, :test do
+  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
+  gem 'byebug'
```
Comment (f3r) [inline Gemfile]: pls add `pry`, `better_errors`, etc

File: Gemfile [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+gem 'jquery-rails'
+# Turbolinks makes following links in your web application faster. Read more: https://github.com/rails/turbolinks
+# gem 'turbolinks'
+# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
+gem 'jbuilder', '~> 2.0'
+# bundle exec rake doc:rails generates the API under doc/api.
+gem 'sdoc', '~> 0.4.0', group: :doc
+
+#gems for translator
+gem 'chinese_pinyin'
```
Comment (f3r) [inline Gemfile:32]: this you should add when you need it, not by default

File: app/assets/javascripts/application.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+//
+// It's not advisable to add code directly here, but if you do, it'll appear at the bottom of the
+// compiled file.
+//
+// Read Sprockets README (https://github.com/rails/sprockets#sprockets-directives) for details
+// about supported directives.
+//
+//= require jquery
+//= require jquery_ujs
+//= require turbolinks
```
Comment (f3r) [inline app/assets/javascripts/application.js]: remove turbolinks

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>MechaAssist</title>
+  <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
+  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
```
Comment (f3r) [inline app/views/layouts/application.html.erb]: turbolinks

File: config/database.yml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+development:
+  <<: *default
+  database: rails_mecha-assist_development
+
+# Warning: The database defined as "test" will be erased and
+# re-generated from your development database when you run "rake".
+# Do not set this db to the same as development or production.
+test:
+  <<: *default
+  database: rails_mecha-assist_development
```
Comment (f3r) [inline config/database.yml:22]: cannot have the same name as development database

File: config/database.yml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+production:
+  <<: *default
+  database: db/rails_mecha-assist_development
```
Comment (f3r) [inline config/database.yml]: for production we are going to use heroku later, leave empty for now

File: Gemfile (hunk 1 above)
Comment (monogut) [inline Gemfile]: fixed

File: Gemfile (hunk 2 above)
Comment (monogut) [inline Gemfile]: added

File: app/assets/javascripts/application.js (hunk 4 above)
Comment (monogut) [inline app/assets/javascripts/application.js]: removed

File: config/database.yml (hunk 6 above)
Comment (monogut) [inline config/database.yml:22]: fixed

**Comment by f3r** [comment]
@monogut is this ready for review?

File: Gemfile [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+gem 'uglifier', '>= 1.3.0'
+# Use CoffeeScript for .coffee assets and views
+gem 'coffee-rails', '~> 4.1.0'
+# See https://github.com/rails/execjs#readme for more supported runtimes
+# gem 'therubyracer', platforms: :ruby
+
+# Use jquery as the JavaScript library
+gem 'jquery-rails'
+# Turbolinks makes following links in your web application faster. Read more: https://github.com/rails/turbolinks
+# gem 'turbolinks'
```
Comment (f3r) [inline Gemfile:25]: remove line instead of commenting it

**Pull request closed without merging**